dev/scripts/generate-tarball.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+# This script generates a tarball of the current Alluxio commit. It does the following:
+# 1. Copy everything except logs/ and dev/ to a work directory
+# 2. Clean out unrevisioned files
+# 3. Compile
+# 4. Use bin/alluxio version to determine the right directory name, e.g. alluxio-1.2.0
+# 5. Tar everything up and put it in dev/scripts/tarballs
+#
```

Comment:
I found it is often very useful to put a usage example here especially a few months later when we get back to the script, it can quickly help us get start. Could you add one? Up to you. approve my review now.